corehq/blobs/migrate.py

```diff
@@ -356,7 +356,7 @@ def get_type_code(doc):
         self.get_type_code = get_type_code
 
     def migrate(self, filename=None, reset=False, max_retry=2, chunk_size=100, **kw):
-        doc_provider = self.get_document_provider()
+        doc_provider = self.get_document_provider(**kw)
```

Comment:
To facilitate my other comment, this will need to pass only arguments from `kw` that are accepted by `get_document_provider`:
```suggestion
        doc_provider_kw = {}
        if "date_range" in kw:
            doc_provider_kw["date_range"] = kw.pop("doc_provider_kw")
        doc_provider = self.get_document_provider(**doc_provider_kw)
```